### ISSUE TITLE
Unable to load 1.16.5 Save in 1.17 versions of game 

### ISSUE THREAD

**Issue by Rufkut** [anegostudios/VintageStory-Issues#2220]
**Game Version:**  v1.17.4 (Stable)
**Platform:** Running on 64 bit Windows with 16 GB RAM 
**Modded:** No 
**SP/MP:** Singleplayer

### Description
The game crashes when I try to load 1.6.5 Save; https://www.mediafire.com/file/2cc4wczt7hew607/LP_-_No_Mods.vcdbs/file. This is my first time updating to a version of 1.17. I downgraded it to 1.16.5 and the save loaded fine with all mods disabled. Afterward, I tried VS version 1.17.0 to 1.17.4 but all attempts failed to load the save.

### How to reproduce
Load save in 1.17.4

### Expected behavior
The save should load and update to 1.17.4.

### Logs
```
27/9/2022 1:42:54 am: Critical error occurred
Loaded Mods: game@1.17.4, creative@1.17.4, survival@1.17.4
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Client.NoObf.ClientMain.ProcessServerReady() in VintagestoryLib\Client\ClientMain.cs:line 1921
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 1138
   at Vintagestory.Client.GuiScreenConnectingToServer.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenConnectingToServer.cs:line 383
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 709
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 613
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(Object sender, FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 125
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OpenTK.GameWindow.RaiseRenderFrame(Double elapsed, Double& timestamp) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 476
   at OpenTK.GameWindow.DispatchRenderFrame() in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 452
   at OpenTK.GameWindow.Run(Double updates_per_second, Double frames_per_second) in C:\Users\Nexrem\Desktop\transfer\opentk\src\OpenTK\GameWindow.cs:line 375
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs)
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries containing Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 27/9/2022 1:36:01 am, Site = , Source = .NET Runtime, Message = Application: Vintagestory.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at Vintagestory.Server.ServerMain.BuildServerAssetsPacket()
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Threading.ThreadHelper.ThreadStart()

 }
--------------
{ TimeGenerated = 27/9/2022 12:44:44 am, Site = , Source = Windows Error Reporting, Message = Fault bucket 1786744555586049135, type 4
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: Vintagestory.exe
P2: 1.16.5.0
P3: 625674b5
P4: StackHash_000e
P5: 0.0.0.0
P6: 00000000
P7: c0000005
P8: PCH_F8_FROM_ntdll+0x000000000009DB54
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER63AF.tmp.mdmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER6B41.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER6B61.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER6B7F.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER6BBE.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Vintagestory.exe_544da8e4f8f54868f72a8aef37119bb2b80b29_1f70e3ae_67fbf2b5-97d4-4242-bab9-b6cdc35ca4a2

Analysis symbol: 
Rechecking for solution: 0
Report Id: 6c991b41-8d88-4204-b17f-74936162a268
Report Status: 268435456
Hashed bucket: d534b5d0ea625c3808cbcab4ec11d46f
Cab Guid: 0 }
--------------
{ TimeGenerated = 27/9/2022 12:44:42 am, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.16.5.0, time stamp: 0x625674b5
Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x00007ffa40c47ae4
Faulting process id: 0x51bc
Faulting application start time: 0x01d8d233906fa308
Faulting application path: C:\Users\cjbox\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: unknown
Report Id: 6c991b41-8d88-4204-b17f-74936162a268
Faulting package full name: 
Faulting package-relative application ID:  }
```

[server-event.txt](https://github.com/anegostudios/VintageStory-Issues/files/9656587/server-event.txt)
[server-main.txt](https://github.com/anegostudios/VintageStory-Issues/files/9656589/server-main.txt)
[server-storyevent.txt](https://github.com/anegostudios/VintageStory-Issues/files/9656590/server-storyevent.txt)
[client-crash.txt](https://github.com/anegostudios/VintageStory-Issues/files/9656591/client-crash.txt)
[client-debug.txt](https://github.com/anegostudios/VintageStory-Issues/files/9656592/client-debug.txt)
[server-debug.txt](https://github.com/anegostudios/VintageStory-Issues/files/9656593/server-debug.txt)

**Comment by tyronx** [comment]
thanks for the report, we are looking into it

**Comment by tyronx** [comment]
You should be able to load up this world again in 1.17.5. Additionally, I actually had to add an optimization to even get to load this world. There are over 400.000 modded blocks in this savegame, which reached my memory limit. I'm not sure how you were able to play this world with mods enabled. Zeekea and expanded foods seem to be one of the major contributors.

**Comment by Rufkut** [comment]
Hmm 400k!? I downloaded 1.17.5 but it still crashed. Would using creative
mode in 1.16.5 to delete modded blocks then updating to 1.17.5 help?

**Comment by Rufkut** [comment]
I tried deleting all the modded blocks that I could find using 1.16.5  but it still failed in 1.17.5.

**Comment by Rufkut** [comment]
The issue was resolved in the recent release, 1.17.6

**Comment by Rufkut** [comment]
Besides occasional lag spikes the game ran smoothly. I deleted Zeekea and
haven't used that mod in a while. I've checked around ingame and haven't
discovered any Zeekea blocks. Is it possible to delete the blocks that are
causing a problem. I'll play without Expanded Foods and mods in general
seeing they cause these issues.
